R/comparePlot.R

```diff
@@ -96,35 +96,156 @@ comparePlot <- function(compareDF,
-                            length(symbolShape) == 4,
-                            length(symbolColor) == 4,
-                            length(symbolFill) == 4,
-                            msg = paste0("All specified symbol arguments must be of length 4, ",
-                                         "including symbolSize, symbolShape, symbolColor, and symbolFill."))
+                           msg = "Need at least two numeric columns in compareDF.")
+    plotType <- tolower(plotType)
+    assertthat::assert_that(plotType %in% c("canvasxpress", "ggplot"),
+                                       msg = "Plot type must be either canvasXpress or ggplot.")
+    if (any(is.null(pThreshold),
+            !is.numeric(pThreshold),
+            length(pThreshold) != 1)) {
+        warning("pThreshold must be a singular value of class numeric. Assigning default value '0.01'.")
+        pThreshold <- 0.01
+    }
+
+    if (!is.null(title) &&
+        !all(is.character(title), length(title) == 1)) {
+        warning("title must be a singular value of class character. Assigning default value 'NULL'.")
+        title <- NULL
+    }
+
+    if (!is.null(xlab) &&
+        !all(is.character(xlab), length(xlab) == 1)) {
+        warning("xlab must be a singular value of class character. Assigning default value 'NULL'.")
+        xlab <- NULL
+    }
+
+    if (!is.null(ylab) &&
+        !all(is.character(ylab), length(ylab) == 1)) {
+        warning("ylab must be a singular value of class character. Assigning default value 'NULL'.")
+        ylab <- NULL
+    }
+
+    if (any(is.null(symbolColor),
+            !is.character(symbolColor),
+            length(symbolColor)  != 4)) {
+        warning("symbolColor must be a vector of 4 character values. Assigning default values 'darkgoldenrod1', 'deepskyblue4', 'red3', 'grey25'.")
+        symbolColor <- c("darkgoldenrod1", "deepskyblue4", "red3", "grey25")
+    }
+
+    if (missing(symbolSize)) {
+        if (plotType == "canvasxpress") {
+            symbolSize  <-  c(10, 10, 10, 4)
+        } else {
+            symbolSize  <-  c(4, 4, 4, 2)
+        }
+    } else if (any(is.null(symbolSize),
+                   !is.numeric(symbolSize),
+                   length(symbolSize)  != 4)) {
+        if (plotType == "canvasxpress") {
+            warning("symbolSize must be a vector of 4 integer values. Assigning canvasXpress default values 10, 10, 10, 4.")
+            symbolSize  <-  c(10, 10, 10, 4)
+        } else {
+            warning("symbolSize must be a vector of 4 integer values. Assigning ggplot default values 4, 4, 4, 2.")
+            symbolSize  <-  c(4, 4, 4, 2)
+        }
+    }
+
+    if (missing(symbolShape)) {
+        if (plotType == "canvasxpress") {
+            symbolShape  <- c("sphere", "sphere", "sphere", "square")
+        } else {
+            symbolShape  <- c("circle filled", "circle filled", "circle filled", "circle small")
+        }
+    } else if (any(is.null(symbolShape),
```

Comment:
refer to the shape validation functions  used in ggplotMDS.R

Reply:
done